pkg/cmd/pulumi/package.go

```diff
@@ -102,7 +111,32 @@ func schemaFromSchemaSource(packageSource string) (*schema.Package, error) {
 		return nil, err
 	}
 	defer p.Close()
-	bytes, err := p.GetSchema(plugin.GetSchemaRequest{})
+
+	var request plugin.GetSchemaRequest
+	if len(args) > 0 {
+		response, err := p.Parameterize(ctx, &pulumirpc.ParameterizeRequest{
+			Parameters: &pulumirpc.ParameterizeRequest_Args{
+				Args: &pulumirpc.ParameterizeRequest_ParametersArgs{
+					Args: args,
+				},
+			},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("parameterize: %w", err)
+		}
+
+		ver, err := semver.ParseTolerant(response.Version)
```

Comment:
Can we take the opportunity to be as strict as possible here?
```suggestion
		ver, err := semver.Parse(response.Version)
```

Reply:
Sure